vscode/src/prompt-builder/index.ts

```diff
@@ -43,6 +48,21 @@ export class PromptBuilder {
 
     private constructor(private readonly tokenCounter: TokenCounter) {}
 
+    private _isCacheEnabled: boolean | undefined
+    private get isCacheEnabled(): boolean {
+        if (this._isCacheEnabled === undefined) {
+            this._isCacheEnabled = logFirstEnrollmentEvent(
+                FeatureFlag.CodyPromptCachingOnMessages,
+                !!storeLastValue(
+                    featureFlagProvider.evaluatedFeatureFlag(
+                        FeatureFlag.SmartApplyContextDataCollectionFlag
```

Comment:
```suggestion
                        FeatureFlag.SmartApplyContextDataCollectionFlag
```

Is this the right flag name?

Reply:
sorry for the mistake here! Updated the code